[clang-tidy][run-clang-tidy] Fix minor shutdown noise

nicovank opened the pull request:
On my new machine, the script outputs some shutdown noise:
```
Ctrl-C detected, goodbye.
Traceback (most recent call last):
  File "/home/nvankempen/llvm-project/./clang-tools-extra/clang-tidy/tool/run-clang-tidy.py", line 626, in <module>
    asyncio.run(main())
  File "/usr/lib/python3.10/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/lib/python3.10/asyncio/base_events.py", line 636, in run_until_complete
    self.run_forever()
  File "/usr/lib/python3.10/asyncio/base_events.py", line 603, in run_forever
    self._run_once()
  File "/usr/lib/python3.10/asyncio/base_events.py", line 1871, in _run_once
    event_list = self._selector.select(timeout)
  File "/usr/lib/python3.10/selectors.py", line 469, in select
    fd_event_list = self._selector.poll(timeout, max_ev)
KeyboardInterrupt
```

This fixes it. Also remove an unused typing import.
Relevant documentation: https://docs.python.org/3/library/asyncio-runner.html#handling-keyboard-interruption

5chmidti reviewed (approved):
Because the commit with asyncio is in llvm-19, could you please add a release note for this fix?

@PiotrZSL wdyt about backporting this? 19.1.0 may be too late, but for 19.2.0 then

nicovank committed:
[clang-tidy][run-clang-tidy] Fix minor shutdown noise

On my new machine, the script outputs some shutdown noise:
```
Ctrl-C detected, goodbye.
Traceback (most recent call last):
  File "/home/nvankempen/llvm-project/./clang-tools-extra/clang-tidy/tool/run-clang-tidy.py", line 626, in <module>
    asyncio.run(main())
  File "/usr/lib/python3.10/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/lib/python3.10/asyncio/base_events.py", line 636, in run_until_complete
    self.run_forever()
  File "/usr/lib/python3.10/asyncio/base_events.py", line 603, in run_forever
    self._run_once()
  File "/usr/lib/python3.10/asyncio/base_events.py", line 1871, in _run_once
    event_list = self._selector.select(timeout)
  File "/usr/lib/python3.10/selectors.py", line 469, in select
    fd_event_list = self._selector.poll(timeout, max_ev)
KeyboardInterrupt
```

This fixes it.

nicovank commented:
Added release note, rebased to trigger CI again.

FWIW I don't remember this happening on any of the ~3 machines I tested on previously, and it only happens on interrupt, not normal execution. If you think this should be backported, let me know if there is anything I can help with.

If/when ready you can go ahead and merge this, I have no commit access.

nicovank commented:
I'm going to go ahead and merge this. IMO this doesn't need backporting because 1. it only happens on abnormal exit where users probably wouldn't be surprised to see a stacktrace anyway, and 2. there is no regression really (check prior to asyncio change also had some noise after Ctrl+C message). But if someone thinks otherwise I'll stamp it 👍.
